FS: Fix user authenticated routes

kyi193 opened the pull request:
Hey guys, this was a hot fix for miscellaneous user authentication issues:

Logout now deletes the token cookie in the browser
Routes that require an authenticated user properly redirects to the sign up page
Sign up and log in pages properly redirects users to the dashboard page if there is an authenticated user

I had a huge issue with merging conflicts with dev with my first PR (fix-routing-and-authentication-on-refresh). That PR has been closed now. The issue was that some of the previous commits (which was not currently fixing this issue), was getting causing all sorts of conflicts with my first PR and I was in rebase purgatory for 6 hours. Instead of trying to merge dev -> the last PR, I decided to pull in the remote dev to my dev first and then manually patch things up so that it works. Hopefully there won't be any merge conflicts this time around but if there are I'm going to need some help. 

I apologize for the last messy commit "Big Hunk", I tried to be as organized as possible but it's 4:30 right now and I need some sleep so we can get waiting room sockets to work :(

kyi193 commented:
Hey guys, so I fixed up my end points so that it uses the passport middleware. I also deleted cors since we won't be needing it anymore. Let me know what you think @skassam21. I didn't put the middleware in logout since all we're doing on logout is deleting the cookie in the browser (correct me if I'm wrong here though!).

users.js (in routes folder)
<img width="434" alt="Screen Shot 2020-11-21 at 12 27 52 PM" src="https://user-images.githubusercontent.com/61174647/99883367-fa858100-2bf4-11eb-8105-50f1a8e6551a.png">
userController.js (all we're doing is returning req.user now since the middleware will handle the jwt verification)
<img width="570" alt="Screen Shot 2020-11-21 at 12 28 32 PM" src="https://user-images.githubusercontent.com/61174647/99883378-125d0500-2bf5-11eb-835e-6c6d63a56fae.png">

skassam21 commented on client/src/Components/Navbar.js:
```diff
@@ -45,7 +46,7 @@ const Navbar = (props) => {
     if (route === '/signup') {
       dispatch({ type: LOGOUT });
-      fetch('/users/logout', { method: 'POST', credentials: 'same-origin' });
+      axios.post('http://localhost:3000/users/logout')
```
We should not be writing localhost:3000 however - it should be /users/logout actually 

I do think we should make this an axios call though, and should be where our other API calls are being made

skassam21 commented on client/src/Components/UserInformation.js:
```diff
@@ -74,11 +74,11 @@ const UserInformation = (props) => {
       history.push('/dashboard');
     } catch (error) {
-      console.error(error);
+      console.log(error);
```
I think we wanted to keep this as an error log

skassam21 commented on client/src/components/LoginSignupWrapper.js:
```diff
@@ -2,34 +2,26 @@ import React, { useContext, useEffect, useCallback } from 'react';
-      });
-    } catch (err) {
-      console.error(err);
+      await axios.get('http://localhost:3000/users/')
```
again we shouldn't be using localhost:3000

skassam21 commented on client/src/routing/PrivateRoute.js:
```diff
@@ -1,11 +1,38 @@
+  const getUserInfo = useCallback(async () => {
+
+    try {
+      const result = await axios.get('http://localhost:3000/users/')
```
again same issue

skassam21 commented on server/controllers/userController.js:
```diff
@@ -54,21 +58,8 @@ function editUser(req, res) {
-    res.status(401).json({ error: 'Token is not valid' });
-  }
+function getUser(req, res) {
+  res.status(200).json({ user: req.user });
```
👍

skassam21 commented on server/app.js:
```diff
@@ -17,7 +16,6 @@ const questions = require('./routes/questions');
 const { json, urlencoded } = express;
 
 var app = express();
-app.use(cors({ origin: 'http://localhost:3000', credentials: true }));
```
👍

skassam21 commented on server/routes/users.js:
```diff
@@ -11,17 +11,12 @@ router.post('/register', userController.register);
-  userController.logout
-);
+// POST /users/logout
+router.post('/logout', userController.logout);
```
this route needs jwt authentication as well

kyi193 commented:
I changed it so that we're using the api functions for axios calls and not use localhost:3000 anymore. I also added the jwt authentication on user logout as well.
